add aws release 20.0.0 (#1204)
giantswarm/releases: aws/v20.0.0/README.md

# :zap: Giant Swarm Release v20.0.0 for AWS :zap:

The new major version is the first release supporting Kubernetes 1.25. 

Please remember that the Pod Security Policies (PSP) are no longer supported in this version and they have to be removed *BEFORE* the upgrade. From now on the Pod Security Standards (PSS) are the default for securing the clusters. Our docs offer additional information about [Pod Security Standards](https://docs.giantswarm.io/advanced/security-policy-enforcement/) as well as a [PSS migration guide](https://docs.giantswarm.io/advanced/security/security-policy-enforcement/cluster-admin-guide/) for cluster administrators.

> This release will also be used as a base for `Giant Swarm Vintage` to Giant Swarm adoption of `Cluster API for AWS` (CAPA) migration. The migration will be scheduled as any other upgrade. We will be reaching out separately to each customer to present the plan as well as the CAPA benefits.

Reviewer: ```suggestion
This release will also be used as a base for the migration from `Giant Swarm Vintage` to `Cluster API for AWS` (CAPA). The migration will be scheduled as any other upgrade. We will be reaching out separately to each customer to present the plan as well as the CAPA benefits.
```

> > **WARNING:** After upgrading to `19.3.0`, it is highly advised to begin removal of all PSPs from the cluster. Kubernetes `v1.25` removes the Pod Security Policy resource from the API, meaning workloads (like Helm charts) which still contain PSPs will fail to install after the upcoming Giant Swarm `v20` release.

Reviewer: ```suggestion
> **WARNING:** After upgrading to `19.3.0`, it is highly advised to begin removal of all customer-managed PSPs from the cluster. Kubernetes `v1.25` removes the Pod Security Policy resource from the API, meaning workloads (like Helm charts) which still contain PSPs will fail to install after the Giant Swarm `v20` upgrade.
```

> > **WARNING:** `Observability-bundle` will be upgraded to `v1.1.0` and it contains breaking changes by simplify configuration for the bundled apps. Please check our [upgrade guide](https://github.com/giantswarm/observability-bundle/blob/main/docs/upgrade.md) or reach out to your Account Engineer for more details.

Reviewer: ```suggestion
> **WARNING:** `observability-bundle` will be upgraded to `v1.1.0`, which contains breaking changes to the configuration for the bundled apps. Please check our [upgrade guide](https://github.com/giantswarm/observability-bundle/blob/main/docs/upgrade.md) or reach out to your Account Engineer for more details.
```

### chart-operator-extensions [1.1.2](https://github.com/giantswarm/chart-operator-extensions/releases/tag/v1.1.2)

#### Fixed
- Move pss values under the global property

### node-exporter [1.19.0](https://github.com/giantswarm/node-exporter-app/releases/tag/v1.19.0)

#### Added
- Add VPA configuration to `node-exporter` app.

### k8s-dns-node-cache-app [2.6.1](https://github.com/giantswarm/k8s-dns-node-cache-app/releases/tag/v2.6.1)

#### Changed
- Configure `gsoci.azurecr.io` as the default container image registry.
- Revert force_tcp option from external DNS zone (#67).

### aws-ebs-csi-driver [2.28.1](https://github.com/giantswarm/aws-ebs-csi-driver-app/releases/tag/v2.28.1)

#### Changed
- Configure `gsoci.azurecr.io` as the default container image registry.

### cluster-autoscaler [1.25.1-gs2](https://github.com/giantswarm/cluster-autoscaler-app/releases/tag/v1.25.1-gs2)

#### Fixed
- Adjusted minimum allowed CPU and memory

### prometheus-blackbox-exporter [0.4.1](https://github.com/giantswarm/prometheus-blackbox-exporter/releases/tag/v0.4.1)

#### Changed
- Configure `gsoci.azurecr.io` as the default container image registry.

> ### security-bundle [1.6.0](https://github.com/giantswarm/security-bundle/releases/tag/v1.6.1)
> 
> #### Changed
> - Update to exception-recommender (app) to v0.1.0.
> - Update to falco (app) to v0.8.0.
> - Update to kyverno-policy-operator (app) version v0.0.7.
> - Update to kyverno (app) version v0.17.2.
> - Update to starboard-exporter (app) version v0.7.8.
> - Update to trivy-operator (app) to v0.7.0.
> - Update to trivy (app) to v0.10.0.
> - Update to kyverno (app) to v0.17.5.
> - Update to exception-recommender (app) to v0.1.1.
> - Update to trivy-operator (app) to v0.7.2.

Reviewer: ```suggestion
### security-bundle [1.6.2](https://github.com/giantswarm/security-bundle/releases/tag/v1.6.2)

#### Changed
- Update to exception-recommender (app) to v0.1.1.
- Update to falco (app) to v0.8.0.
- Update to kyverno-policy-operator (app) version v0.0.7.
- Update to kyverno (app) version v0.17.6.
- Update to starboard-exporter (app) version v0.7.8.
- Update to trivy-operator (app) to v0.7.2.
- Update to trivy (app) to v0.10.0.
- Update to kyverno (app) to v0.17.5.
```

> **WARNING:** After upgrading to `19.3.0`, it is highly advised to begin removal of all PSPs from the cluster. Kubernetes `v1.25` removes the Pod Security Policy resource from the API, meaning workloads (like Helm charts) which still contain PSPs will fail to install after the upcoming Giant Swarm `v20` release.

> > **WARNING:** `observability-bundle` will be upgraded to `v1.1.0`, which contains breaking changes to the configuration for the bundled apps. Please check our [upgrade guide](https://github.com/giantswarm/observability-bundle/blob/main/docs/upgrade.md) or reach out to your Account Engineer for more details.

Reviewer 2: ```suggestion
> **WARNING:** `observability-bundle` will be upgraded to `v1.1.0`, which contains breaking changes to the configuration for the bundled apps to simplify configuration, especially for GitOps. Please check our [upgrade guide](https://github.com/giantswarm/observability-bundle/blob/main/docs/upgrade.md) or reach out to your Account Engineer for more details.
```

Reviewer 2: @T-Kukawka do you think we could bump the observability bundle to 1.2.4?

Reviewer 3: if we are fine that it does not break anything, maybe worth to test real quick? then yes.

Reviewer 2: It should not but can we do a test release quick? :D

Reviewer 3: @QuentinBisson u can just take that release.yaml, apply it on the Test MC or on `garfish` named e.g. 20.0.1, adjust your app in the yaml and deploy the cluster with that release basically - we do not cut test releases

Reviewer 2: sure I'm aware. I'll test tomorrow because I'm supposed to be off today :D

Reviewer 3: sure 👍 no worries

Reviewer 2: Marie ran a test and it worked like a charm with version 1.3.0 even :D

Reviewer 2: https://github.com/giantswarm/releases/pull/1217

Reviewer 3: approved
